src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+type TLocalStorageValue = string | boolean | number;
+
+interface IUseLocalStorageProps<T extends TLocalStorageValue> {
+  localStorageKey: string;
+  initialValue: T;
+  localStorageValueParser?: (localStorageValue: string) => T;
```

Comment:
If `initialValue` is mandatory, then `localStorageValueParser` could be theoretically determined from it = if `initialValue` is `boolean`, then `booleanLocalStorageValueParser` could be used automatically. 

But I am not sure about possible unwanted side-effects as it can make this hook less reusable and less explicit.